pymeasure/instruments/attocube/anc300.py

```diff
@@ -0,0 +1,227 @@
+
+    capacity = Instrument.measurement("getc",
+                                      """ Saved capacity value in nF of the
+                                      axis.""",
+                                      get_process=extract_float)
+
+    stepu = Instrument.setting(
+            "stepu %d",
+            """ Step upwards for N steps. Mode must be 'stp' and N must be
+            positive.""",
+            validator=strict_range, values=[0, inf])
+
+    stepd = Instrument.setting(
+            "stepd %d",
+            """ Step downwards for N steps. Mode must be 'stp' and N must be
+            positive.""",
+            validator=strict_range, values=[0, inf])
+
+    def __init__(self, axis, controller):
+        self.axis = str(axis)
+        self.controller = controller
+
+    def _add_axis_id(self, command):
+        """ add axis id to a command string at the correct position after the
+        initial command, but before a potential value
+
+        :param command: command string
+        :returns: command string with added axis id
+        """
+        cmdparts = command.split()
+        cmdparts.insert(1, self.axis)
+        return ' '.join(cmdparts)
+
+    def ask(self, command, **kwargs):
+        return self.controller.ask(self._add_axis_id(command), **kwargs)
+
+    def write(self, command, **kwargs):
+        return self.controller.write(self._add_axis_id(command), **kwargs)
+
+    def values(self, command, **kwargs):
+        return self.controller.values(self._add_axis_id(command), **kwargs)
+
+    def stop(self):
+        """ Stop any motion of the axis """
+        self.write('stop')
+
+    def move(self, steps, gnd=True):
+        """ Move 'steps' steps in the direction given by the sign of the
+        argument. This method will change the mode of the axis automatically
+        and ground the axis on the end if 'gnd' is True. The method returns
+        only when the movement is finished.
+
+        :param steps: finite integer value of steps to be performed. A positive
+        sign corresponds to upwards steps, a negative sign to downwards steps.
+        :param gnd: bool, flag to decide if the axis should be grounded after
+        completion of the movement
+        """
+        self.mode = 'stp'
+        # perform the movement
+        if steps > 0:
+            self.stepu = steps
+        elif steps < 0:
+            self.stepd = abs(steps)
+        else:
+            pass  # do not set stepu/d to 0 since it triggers a continous move
+        # wait for the move to finish
+        self.write('stepw')
+        if gnd:
+            self.mode = 'gnd'
+
+    def measure_capacity(self):
+        """ Obtains a new measurement of the capacity. The mode of the axis
+        returns to 'gnd' after the measurement.
+
+        :returns capacity: the freshly measured capacity in nF.
+        """
+        self.mode = 'cap'
+        # wait for the measurement to finish
+        self.ask('capw')
```

Comment:
do you want to return the mode to the previously set one after finishing?

Reply:
The controller returns the mode automatically after a single measurement